File: apps/api/app/schemas/account.py
from pydantic import BaseModel, ConfigDict


class AccountCreate(BaseModel):
    account_name: str
    account_type: str
    permission_level: str
    via_bastion: bool
    mfa_enabled: bool


class AccountRead(AccountCreate):
    model_config = ConfigDict(from_attributes=True)

    id: int

File: packages/contracts/src/account.ts
import { z } from "zod";

export const accountSchema = z.object({
  id: z.string(),
  employeeId: z.string().optional(),
  accountName: z.string(),
  accountType: z.enum(["human", "system", "shared", "temporary", "api"]),
  systemId: z.string().optional(),
  permissionLevel: z.enum(["read", "operate", "admin"]),
  viaBastion: z.boolean().default(false),
  mfaEnabled: z.boolean().default(false)
});

export type Account = z.infer<typeof accountSchema>;
